### PR TITLE
[Bugfix][2646] Ability to retreat when player has no towns

### PR DESCRIPTION
**Pull request by tau3**
Hello!

This is a fix for #2646. It needs to be checked before merge. Can I have a save file to reproduce?

### CODE REVIEW

**Comment by Branikolog** [comment]
[BUG.zip](https://github.com/ihhub/fheroes2/files/5856540/BUG.zip)

**Comment by tau3** [comment]
@Branikolog thank you! 

I was able to retreat after being attacked by the nearest hostile hero. 👌

File: src/fheroes2/battle/battle_arena.cpp [hunk 1]
```diff
@@ -731,8 +731,13 @@ bool Battle::Arena::CanSurrenderOpponent( int color ) const
 
 bool Battle::Arena::CanRetreatOpponent( int color ) const
 {
-    const HeroBase * hero = army1->GetColor() == color ? army1->GetCommander() : army2->GetCommander();
-    return hero && hero->isHeroes() && NULL == hero->inCastle() && world.GetKingdom( hero->GetColor() ).GetCastles().size();
```
Comment (ihhub) [inline src/fheroes2/battle/battle_arena.cpp:735]: Could we please just remove `&& world.GetKingdom( hero->GetColor() ).GetCastles().size()` for the whole pull request changes?

**Review by ihhub — CHANGES_REQUESTED** [review]
Hi @tau3 , could you please check proposed approach?

**Comment by ihhub** [comment]
@tau3 thank you very much for the fix!

**Pull request merged**